easy-connect.cpp

```diff
@@ -0,0 +1,277 @@
+/*
+ * FILE: easy-connect.cpp
+ *
+ * Copyright (c) 2015 - 2017 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed.h"
+#include "easy-connect.h"
+
+/*
+ * Instantiate the configured network interface
+ */
+#if MBED_CONF_APP_NETWORK_INTERFACE == WIFI_ESP8266
+#include "ESP8266Interface.h"
+#define WIFI_TYPE "ESP8266"
+
+#ifdef MBED_CONF_APP_ESP8266_DEBUG
+ESP8266Interface wifi(MBED_CONF_APP_ESP8266_TX, MBED_CONF_APP_ESP8266_RX, MBED_CONF_APP_ESP8266_DEBUG);
```

Comment:
We are relying here on macros that require them to be defined in the applications `mbed_app.json`

Should we instead do this properly and have `mbed_lib.json` within this easy-connect library?

Same goes for all `MBED_CONF_APP_*` macros.

This module should be usable even when application does not have `mbed_app.json` at all.

Reply:
They need to have the mbed_app.json anyway for choosing the network, so I'm not seeing the point necessarily in that.

Reply:
The point is that these are configuration variables for Easy connect, not for the application.

Now your module does not compile if application is not defining these in `mbed_app.json` I don't think it is acceptable.